compose-lint-checks/src/main/java/slack/lint/compose/M2ApiDetector.kt

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2022 Salesforce, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.compose
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity.IGNORE
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.TextFormat
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UImportStatement
+import org.jetbrains.uast.UQualifiedReferenceExpression
+import org.jetbrains.uast.UResolvable
+import slack.lint.compose.util.Priorities.NORMAL
+import slack.lint.compose.util.sourceImplementation
+
+internal class M2ApiDetector : Detector(), SourceCodeScanner {
+
+  override fun getApplicableUastTypes() =
+    listOf<Class<out UElement>>(
+      UCallExpression::class.java,
+      UImportStatement::class.java,
+      UQualifiedReferenceExpression::class.java,
+    )
+
+  override fun createUastHandler(context: JavaContext) =
+    object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) = checkNode(node)
```

Comment:
Will the call expression contain the package name? It wouldn't in ktlint, but I guess it will in Lint? If so, I say we don't check the imports.

Having this handle imports means that devs have to use `@file:SuppressLint` to suppress the import warnings, and they're annoying 😆

Reply:
yah you can't since it's in the android jar itself. Remove imports [88b5bf1](https://github.com/slackhq/compose-lints/pull/75/commits/88b5bf104306465c16a6799542d88a9b4739e68c)